screens/ChooseLocationScreen.tsx

```diff
@@ -171,30 +168,7 @@ export const ChooseLocationScreen = ({
           ) : null}
         </MapView>
         <View style={styles.searchInput}>
-          <TextInput
-            ref={searchAddressRef}
-            multiline={false}
-            dense={true}
-            scrollEnabled={true}
-            style={{ height: 60 }}
-            blurOnSubmit={true}
-            outlineColor={theme.colors.disabled}
-            mode="outlined"
-            autoCorrect={false}
-            placeholder="Caută"
-            right={
-              <TextInput.Icon name="magnify" color={theme.colors.placeholder} />
-            }
-            returnKeyType="search"
-            onSubmitEditing={async ({ nativeEvent: { text } }) => {
-              setSelectedAddress(text);
-              searchForAddress(text);
-            }}
-            onChangeText={text => {
-              setSelectedAddress(text);
-            }}
-            value={selectedAddress}
-          />
+          <SearchableLocationDropdown mapRef={mapRef} />
         </View>
```

Comment:
you can use `forwardRef` to have more control about where to inject the `ref` prop.
https://react.dev/reference/react/forwardRef